tests/core/api/test_auth.py

```diff
@@ -0,0 +1,35 @@
+import time
+
+
+def test_token_create_decode(client, settings):
+    res = client.post(
+        f"{settings.api_root}/auth/login",
+        json={"username": "abcd", "password": "abcd", "plugin": "dummy"},
+    )
+    assert res.status_code == 201
+
+    headers = {"Authorization": f"Bearer {res.json()['token']}"}
+    res2 = client.get(f"{settings.api_root}/events", headers=headers)
+    assert res2.status_code == 200
+
+
+def test_token_expired(settings, short_session, client):
+    res = client.post(
+        f"{settings.api_root}/auth/login",
+        json={"username": "abcd", "password": "abcd", "plugin": "dummy"},
+    )
+    assert res.status_code == 201
+
+    time.sleep(short_session * 60 + 1)
+
+    headers = {"Authorization": f"Bearer {res.json()['token']}"}
+    res2 = client.get(f"{settings.api_root}/events", headers=headers)
+    assert res2.status_code == 401
+    assert "expired" in res2.json()["detail"].lower()
+
+
+def test_token_invalid(client, settings):
```

Comment:
In the test previously in place, we used `@pytest.mark.parametrize` to separate [test procedures](https://github.com/ispyb/py-ispyb/blob/37fe33a8b3fbf1d9702fe402c0baa48afa8662a5/tests/core/full_db/functional/test_authorization.py) (i.e. create token, then send request, then check status code and result) and test [data](https://github.com/ispyb/py-ispyb/blob/37fe33a8b3fbf1d9702fe402c0baa48afa8662a5/tests/core/full_db/functional/data/authorization.py) (i.e. valid/invalid token, endpoint to hit, expected status code, expected result).
This allowed us to easily run the same test procedure multiple times with different inputs and expected outputs.

Why not keep it here?

Reply:
These aren't quite the same tests as you were running before, these are authentication. They are testing that the low level token generation is working properly. 

I didn't add any authorisation tests yet.

However requesting an endpoint and checking a status code could be parameterised, using the auth_client as in the `test _events` further below. Sure, this could be applied to authorisation tests.

Reply:
I've made clearer the type of tests by renaming the file